app/src/main/java/org/kiwix/kiwixmobile/webserver/ZimHostFragment.kt

```diff
@@ -132,9 +137,49 @@ class ZimHostFragment : BaseFragment(), ZimHostCallbacks, ZimHostContract.View {
       }
     }
 
-    startServerButton.setOnClickListener { startStopServer() }
+    startServerButton.setOnClickListener {
+      if (Build.VERSION.SDK_INT > Build.VERSION_CODES.P) {
+        if (checkCoarseLocationAccessPermission()) {
+          startStopServer()
+        }
+      } else {
+        startStopServer()
+      }
+    }
   }
 
+  private fun checkCoarseLocationAccessPermission(): Boolean =
+    if (ContextCompat.checkSelfPermission(
+        requireActivity(),
+        Manifest.permission.ACCESS_COARSE_LOCATION
+      ) == PackageManager.PERMISSION_DENIED
+    ) {
+      when {
+        ActivityCompat.shouldShowRequestPermissionRationale(
+          requireActivity(),
+          Manifest.permission.ACCESS_COARSE_LOCATION
+        ) -> {
+          alertDialogShower.show(
+            KiwixDialog.LocationPermissionRationaleOnHostZimFile,
+            {
+              ActivityCompat.requestPermissions(
+                requireActivity(), arrayOf(Manifest.permission.ACCESS_COARSE_LOCATION),
+                PERMISSION_REQUEST_CODE_COARSE_LOCATION
+              )
+            })
+        }
+        else -> {
+          ActivityCompat.requestPermissions(
```

Comment:
duplicated code in both branches, extract a method

Reply:
@macgills is this right? or do we have an extension function for this?
```
  override fun onRequestPermissionsResult(
    requestCode: Int,
    permissions: Array<out String>,
    grantResults: IntArray
  ) {
    super.onRequestPermissionsResult(requestCode, permissions, grantResults)
    if (grantResults[0] == PackageManager.PERMISSION_GRANTED) {
      if (requestCode == PERMISSION_REQUEST_CODE_COARSE_LOCATION) {
        startStopServer()
      }
    }
  }
```